elide-core/src/main/java/com/yahoo/elide/utils/coerce/CoerceUtil.java

```diff
@@ -106,19 +108,41 @@ public Converter lookup(Class<?> sourceType, Class<?> targetType) {
     /**
      * Initialize this classloader if necessary
      */
-    private static void initializeClassLoaderIfNecessary() {
-        ClassLoader currentClassLoader = Thread.currentThread().getContextClassLoader();
-        if (INITIALIZED_CLASSLOADERS.contains(currentClassLoader)) {
+    private static void initializeCurrentClassLoaderIfNecessary() {
+        if (isCurrentClassLoaderInitialized()) {
             return;
         }
         BeanUtilsBean.setInstance(BEAN_UTILS_BEAN_INSTANCE);
-        markClassLoaderAsInitialized();
+        markCurrentClassLoaderAsInitialized();
     }
 
     /**
      * Mark the current class loader as initialized.
      */
-    private static void markClassLoaderAsInitialized() {
-        INITIALIZED_CLASSLOADERS.add(Thread.currentThread().getContextClassLoader());
+    private static void markCurrentClassLoaderAsInitialized() {
+        INITIALIZED_CLASSLOADERS.add(new WeakReference<>(Thread.currentThread().getContextClassLoader()));
+    }
+
+    /**
+     * Method to check if classloader has been loaded or not.
+     *
+     * <strong>NOTE:</strong> The number of class loaders is expected to be <em>small enough</em>, that this
+     * scan per request is negligible.
+     *
+     * @return True if classloader is loaded, false otherwise.
+     */
+    private static boolean isCurrentClassLoaderInitialized() {
```

Comment:
This would all be handled automatically if we used `WeakHashMap` and be more efficient since the Reference queue would be polled for any expired weak references.

Reply:
Even better
```java
    private static final Set<ClassLoader> INITIALIZED_CLASSLOADERS =
            Collections.newSetFromMap(new MapMaker().weakKeys().makeMap());
```